data-rights-protocol.md

```diff
@@ -0,0 +1,232 @@
+
+This is the Data Rights Exercise endpoint which End-Users and Authorized Agents can use to exercise enumerated data rights.
+
+```
+{
+  "meta": {
+    "version": "0.3"
+  },
+  "regulatory_authority": "ccpa",
+  "exercise": [
+    "do_not_sell"
+  ],
+  "identity": <jwt>... ,
+  "status_callback": "https://dsr-agent.example.com/update_status"
+}
+```
+
+- `meta` MUST contain only a single key `version` which contains a string referencing the current protocol version “0.3”.
+- `regulatory_authority` MUST contain a string referencing the regulation under which the Data Request is being taken under
+- `exercise` MUST contain a list of rights to exercise.
+- `identity` MUST contain an [RFC7515 JWT](https://datatracker.ietf.org/doc/html/rfc7515) conforming to one of the following specifications:
+  - a string containing a JWT serialized in the Compact Serialization format [RFC7515 Section 3.1]
+  - a document object containing a JWT serialized in the JSON Serialization formation [RFC7515 Section 3.2]
+- `status_callback` MAY be specified with a URL that the Status Callback can be sent to.
+
+See section 3.04 regarding identity encapsulation.
+
+#### 2.02.1 `POST /exercise` Response
+
+Responses to this request MUST adhere to the [Exercise Status Schema]().
+
+### 2.03 `GET /status` ("Data Rights Status" endpoint)
+
+This is the Data Rights Status endpoint which End-Users and Authorized Agents can use to query for the status of an existing data rights request. Requests to this endpoint MUST provide a single URL parameter request_id which is the Request ID for the Data Rights Request.
+
+#### 2.03.1 `GET /status` Response
+
+Responses to this request MUST adhere to the [Exercise Status Schema]().
+
+
+### 2.04 `POST $status_callback` ("Data Rights Status Callback" endpoint)
+
+The Status Callback endpoint SHOULD be implemented by Authorized Agents which will be exercising data rights for multiple users. This endpoint exists to remove the need for Authorized Agents to query the Data Rights Status endpoint and instead allow a “push model” where AAs are notified when a request's status changes. The destination for a Status Callback URL is specified in the initial [Data Rights Request]().
+
+The request body MUST adhere to the Exercise Status Schema. 
+
+#### 2.04.1 `POST $status_callback` Response
+
+Covered Business SHOULD make a best effort to ensure that a 200 status is recorded for the most recent status update. The body of the callback's response SHOULD be treated as opaque and not be considered for parsing by the Covered Business.
+
+### 2.05 `POST /cancel` ("Data Rights Cancel" endpoint)
+
+[we still need to reason about this, there are some open questiosn I haven't spent much time thinking about on how to line up "revoking consent" with "canceling request" between DNS and access/deletion]
+
+```
+{
+  "request_id": "c789ff35-7644-4ceb-9981-4b35c264aac3",
+  "reason": "i don't want my account deleted"
```

Comment:
this is in a grab-bag of "nice to have" optional fields that i had in mind. feel free to redline